book/core-concepts/partitioning.md

```diff
@@ -106,7 +106,7 @@ func (us *UpdateStock) Compute(data interface{}, state interface{}) (interface{}
 
 ### Partition Key
 
-Currently, the partition keys for a particular partition need to be defined along with the application. The specific details of keys vary between the different language APIs, but they are typically an object of some type that can support comparison and hashing. In the stock example, the partition key would be based on the symbol name (a string). All of the expected stock symbols are passed to the application setup code.
+The partition keys for a particular partition do not need to be defined along with the application. If you don't define your keys ahead of time, they will be automatically generated as needed, while your application is running. If you do choose to define your keys in your application, not that the specific details of keys vary between the different language APIs. In the stock example, the partition key would be based on the symbol name (a string). All of the expected stock symbols are passed to the application setup code.
```

Comment:
I think you should mention that the keys are derived from inputs using the partition function and state partitions are generated for keys that haven't been seen yet.